gobblin-runtime/src/main/java/gobblin/scheduler/JobScheduler.java

```diff
@@ -388,7 +391,156 @@ private void scheduleGeneralConfiguredJobs()
+    }
+
+    public void loadNewJobConfigAndHandleNewJob(Path path, Action action) {
+      // Load the new job configuration and schedule the new job
+
+      String customizedInfo = "";
+      try {
+        Properties jobProps =
+            SchedulerUtils.loadGenericJobConfig(JobScheduler.this.properties, path, jobConfigFileDirPath);
+        System.err.println("New finding jobProps has size : " + jobProps.size());
+
+        switch (action) {
+          case SCHEDULE:
+            boolean runOnce = Boolean.valueOf(jobProps.getProperty(ConfigurationKeys.JOB_RUN_ONCE_KEY, "false"));
+            scheduleJob(jobProps, runOnce ? new RunOnceJobListener() : new EmailNotificationJobListener());
+            customizedInfo = "schedule";
+            break;
+          case UNSCHEDULE:
+            String jobName = jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY);
+            unscheduleJob(jobName);
+            customizedInfo = "unschedule";
+            break;
+          case RESCHEDULE:
+            rescheduleJob(jobProps);
+            customizedInfo = "reschedule";
+            break;
+          default:
+            break;
+        }
+
+      } catch (ConfigurationException | IOException e) {
+        LOG.error("Failed to load from job configuration file " + path.toString(), e);
+      } catch (JobException je) {
+        LOG.error("Failed to " + customizedInfo + " new job loaded from job configuration file " + path.toString(), je);
+      }
+
+    }
+
+    public void loadNewCommonConfigAndHandleNewJob(Path path, Action action) {
+      String customizedInfoAction = "";
+      String customizedInfoResult = "";
+      try {
+        for (Properties jobProps : SchedulerUtils.loadGenericJobConfigs(JobScheduler.this.properties, path,
+            jobConfigFileDirPath)) {
+          try {
+            switch (action) {
+              case RESCHEDULE:
+                rescheduleJob(jobProps);
+                customizedInfoAction = "schedule";
+                customizedInfoResult = "change";
+                break;
+              case UNSCHEDULE:
+                String jobName = jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY);
+                unscheduleJob(jobName);
+                customizedInfoAction = "unschedule";
+                customizedInfoResult = "deletion";
+                break;
+              default:
+                break;
+            }
+          } catch (JobException je) {
+            LOG.error("Failed to " + customizedInfoAction + " job reloaded from job configuration file "
+                + jobProps.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_PATH_KEY), je);
+          }
+        }
+      } catch (ConfigurationException | IOException e) {
+        LOG.error(
+            "Failed to reload job configuration files affected by " + customizedInfoResult + " to " + path.toString(),
+            e);
+      }
+    }
+
+    @Override
+    public void onFileCreate(Path path) {
```

Comment:
In the original Yinan's implementation there's no such routine. 
I was wondering if we need this, as we may make any configuration file mutable later?